recipes/libelfin/all/CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+set(elf_headers
+    source_subfolder/elf/common.hh
+    source_subfolder/elf/data.hh
+    source_subfolder/elf/elf++.hh)
+file(GLOB_RECURSE dwarf_sources source_subfolder/dwarf/*.cc)
+set(dwarf_headers
+    source_subfolder/dwarf/data.hh
+    source_subfolder/dwarf/dwarf++.hh
+    source_subfolder/dwarf/small_vector.hh)
+
+add_custom_command(
+    OUTPUT source_subfolder/elf/to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo '// Automatically generated' > to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo '// DO NOT EDIT' >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo '\#include \"data.hh\"' >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo '\#include \"to_hex.hh\"' >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo 'ELFPP_BEGIN_NAMESPACE' >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+    COMMAND ${PYTHON_EXECUTABLE} enum-print.py -u --hex --no-type --mask shf --mask pf -x loos -x hios -x loproc -x hiproc < data.hh >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo 'ELFPP_END_NAMESPACE' >> to_string.cc
+    DEPENDS source_subfolder/elf/enum-print.py source_subfolder/elf/data.hh
+    WORKING_DIRECTORY source_subfolder/elf)
+
+add_custom_command(
+    OUTPUT source_subfolder/dwarf/to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo '// Automatically generated' > to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo '// DO NOT EDIT' >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo '\#include \"internal.hh\"' >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo 'DWARFPP_BEGIN_NAMESPACE' >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo >> to_string.cc
+    COMMAND ${PYTHON_EXECUTABLE} ../elf/enum-print.py < dwarf++.hh >> to_string.cc
+    COMMAND ${PYTHON_EXECUTABLE} ../elf/enum-print.py -s _ -u --hex -x hi_user -x lo_user < data.hh >> to_string.cc
+    COMMAND ${CMAKE_COMMAND} -E echo 'DWARFPP_END_NAMESPACE' >> to_string.cc
+    DEPENDS source_subfolder/elf/enum-print.py source_subfolder/dwarf/data.hh
+    WORKING_DIRECTORY source_subfolder/dwarf)
+
+add_library(elf++ ${elf_sources} source_subfolder/elf/to_string.cc)
+set_target_properties(elf++ PROPERTIES
+    PUBLIC_HEADER "${elf_headers}"
+    VERSION ${CONAN_PACKAGE_VERSION})
+
+add_library(dwarf++ ${dwarf_sources} source_subfolder/dwarf/to_string.cc)
+set_target_properties(dwarf++ PROPERTIES
+    PUBLIC_HEADER "${dwarf_headers}"
+    VERSION ${CONAN_PACKAGE_VERSION})
+
+install(TARGETS elf++
+    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib
+    RUNTIME DESTINATION bin
+    PUBLIC_HEADER DESTINATION include/libelfin/elf)
+install(TARGETS dwarf++
+    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib
+    RUNTIME DESTINATION bin
+    PUBLIC_HEADER DESTINATION include/libelfin/dwarf)
```

Comment:
```suggestion
install(TARGETS elf++
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libelfin/elf)
install(TARGETS dwarf++
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libelfin/dwarf)
```

Reply:
`CMAKE_INSTALL_*` variables are available only with GNUInstallDirs module. Why to use it? It's wordy and doing just the same.

Looking at other recipes I don't see there is a standard. Need some justification for the suggestion.

Reply:
For a recipe it's ok to hardcode the paths I guess.
Have you looked in upstreaming these patches?
Then @intelligide 's suggestion + `include(GNUInstallDirs)` would be super.

Reply:
Conan already set these variables to the default recipe paths. (See [here](https://github.com/conan-io/conan/blob/edb3a4a1f7739027c0665749b82b1cc473453ab6/conans/client/build/cmake_flags.py#L356)). Adding `include(GNUInstallDirs)` is better.

Reply:
The original repository doesn't have CMake files. Yes, we contacted the authors of the library about adding CMake support, but we didn't get any reply so far.